component/azstorage/azstorage.go

```diff
@@ -276,6 +295,8 @@ func (az *AzStorage) StreamDir(options internal.StreamDirOptions) ([]*internal.O
 
 	log.Debug("AzStorage::StreamDir : Retrieved %d objects with %s marker for Path %s", len(new_list), options.Token, path)
 
+	AzStatsCollector.AddStats(az.Name(), "StreamDir", path, true, map[string]interface{}{"Count": len(new_list)})
```

Comment:
for streamdir we shall also put the name of the directory being listed.

Reply:
The **path** will indicate the directory being listed. If the listing operation is done on the root, path will be an empty string. Otherwise, it will be the relative path of the folder being listed.